fuzz/fuzz_targets/fuzz_int_set.rs

```diff
@@ -0,0 +1,673 @@
+#![no_main]
+//! A correctness fuzzer that checks all of the public API methods of IntSet.
+//!
+//! This fuzzer exercises the public API of IntSet and compares the results to the same operations run
+//! on a BTreeSet. Any differences in behaviour and/or set contents triggers a panic.
+//!
+//! The fuzzer input data is interpretted as a series of op codes which map to the public api methods of IntSet.
+//!
+//! Note: currently only inclusive mode IntSet's are tested.
+
+use std::cmp::max;
+use std::ops::Bound::Excluded;
+use std::ops::Bound::Included;
+use std::ops::RangeInclusive;
+use std::{collections::BTreeSet, error::Error};
+
+use int_set::IntSet;
+use libfuzzer_sys::fuzz_target;
+
+const OPERATION_COUNT: usize = 7_500;
+
+// TODO(garretrieger): allow inverted sets to be accessed.
+// TODO(garretrieger): allow a limited domain set to be accessed.
+
+struct Input<'a> {
+    // The state includes 2 of each type of sets to allow us to test out binary set operations (eg. union)
+    int_set: &'a mut IntSet<u32>,
+    btree_set: &'a mut BTreeSet<u32>,
+}
+
+impl Input<'_> {
+    fn from<'a>(int_set: &'a mut IntSet<u32>, btree_set: &'a mut BTreeSet<u32>) -> Input<'a> {
+        Input { int_set, btree_set }
+    }
+}
+
+trait Operation {
+    fn size(&self, set_len: usize) -> usize;
+    fn operate(&self, input: Input, other: Input);
+}
+
+/* ### Insert ### */
```

Comment:
Seems self-evident, no comment required?

Reply:
I found these useful when working on this to visually break up the blocks of op specific code.